test/jdk/java/awt/print/PrinterJob/PrintNullString.java

```diff
@@ -21,308 +21,160 @@
-        } catch (NullPointerException e) {
-          g.drawString("caught expected NPE for null iterator, float", 20, 160);
-        } /*catch (Exception e) {
-          g.drawString("FAILURE: unexpected exception for null iterator, float",
-                        20, 160);
-        } */
-
-        try {
-             g.drawString(emptyIterator, 20, 180);
-             g.drawString("FAILURE: No IAE for empty iterator, float",
-                           20, 180);
-        } catch (IllegalArgumentException e) {
-          g.drawString("caught expected IAE for empty iterator, float",
-                        20, 180);
-        } /*catch (Exception e) {
-          g.drawString("FAILURE: unexpected exception for empty iterator, float",
-                       20, 180);
-        } */
-    }
-
-     public Dimension getPreferredSize() {
-        return new Dimension(450, 250);
-    }
- }
-
-}
-
-class Sysout
- {
-   private static TestDialog dialog;
-
-   public static void createDialogWithInstructions( String[] instructions )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      dialog.printInstructions( instructions );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-
-   public static void printInstructions( String[] instructions )
-    {
-      dialog.printInstructions( instructions );
+public class PrintNullString extends Frame {
+    private static final String INSTRUCTIONS =
+            "This test should print a page which contains the same\n" +
+            "text messages as in the test window on the screen\n" +
+            "The messages should contain only 'OK' and 'expected' messages\n" +
+            "If the page fails to print, but there were no exceptions\n" +
+            "then the problem is likely elsewhere (ie your printer)";
+
+    public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(PrintNullString::new)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build()
+                .awaitAndCheck();
     }
 
-
-   public static void println( String messageIn )
-    {
-      dialog.displayMessage( messageIn );
+    public PrintNullString() {
+        super("PrintNullString");
+
+        TextCanvas c = new TextCanvas();
+        add("Center", c);
+
+        JButton b = new JButton("Print");
```

Comment:
```suggestion
        Button b = new Button("Print");
```
Better use `Button` with other AWT components.